### PR TITLE
refactor: Run textarea resizing asynchronously (WEBAPP-5548)

### PR DESCRIPTION
**Pull request by atomrc**
Will prevent reflow operations to block the thread while typing in the textarea

### CODE REVIEW

File: app/script/view_model/bindings/CommonBindings.js [hunk 1]
```diff
@@ -175,10 +176,10 @@ ko.bindingHandlers.resize = (function() {
           {
             event: {
               focus() {
-                resizeTextarea(element);
+                throttledResizeTextarea(element);
```
Comment (gregor) [inline app/script/view_model/bindings/CommonBindings.js]: Any reason why these are not single line arrow functions? `focus: () => throttledResizeTextarea(element),`

Comment (atomrc) [reply app/script/view_model/bindings/CommonBindings.js]: The only reason I can think of is legacy.

File: app/script/view_model/bindings/CommonBindings.js [hunk 2]
```diff
@@ -191,8 +188,9 @@ ko.bindingHandlers.resize = (function() {
       const params = ko.unwrap(valueAccessor()) || {};
       triggerValue = params.triggerValue;
       syncElement = document.querySelector(params.syncElement);
+      throttledResizeTextarea = _.throttle(resizeTextarea, 100, {leading: !params.delayedResize});
```
Comment (atomrc) [inline app/script/view_model/bindings/CommonBindings.js]: There might a problem here, I should create the throttled version in the `init` function instead

Comment (gregor) [reply app/script/view_model/bindings/CommonBindings.js]: Yeah, that actually makes a lot of sense.

File: app/script/view_model/bindings/CommonBindings.js [hunk 3]
```diff
@@ -188,11 +186,10 @@ ko.bindingHandlers.resize = (function() {
     },
 
     update(element, valueAccessor) {
-      const params = ko.unwrap(valueAccessor()) || {};
-      triggerValue = params.triggerValue;
-      syncElement = document.querySelector(params.syncElement);
```
Comment (atomrc) [inline app/script/view_model/bindings/CommonBindings.js:193]: Another perf improvement here, we now don't query the DOM on every single input (it is done when the binding is initiated)

**Pull request merged**